### PR TITLE
refactor: update `blas/ext/base/dcusum` to follow current project conventions

### LINKED ISSUE #1471: [RFC]: refactor `blas/ext/base/dcusum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dcusum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dcusum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AuenKr**
Resolves #1471.

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors `@stdlib/blas/ext/base/dcusum` to follow current projects convention

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1471 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by AuenKr** [comment]
@Pranavchiku, can you review the PR?

File: lib/node_modules/@stdlib/blas/ext/base/dcusum/README.md [hunk 1]
```diff
@@ -65,18 +65,15 @@ The function has the following parameters:
 -   **y**: output [`Float64Array`][@stdlib/array/float64].
 -   **strideY**: index increment for `y`.
 
-The `N` and `stride` parameters determine which elements in `x` and `y` are accessed at runtime. For example, to compute the cumulative sum of every other element in `x`,
+The `N` and `stride` parameters determine which elements in the strided input and output array are accessed at runtime. For example, to compute the cumulative sum of every other element in strided input array,
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusum/README.md]:
```suggestion
The `N` and `stride` parameters determine which elements in the strided arrays are accessed at runtime. For example, to compute the cumulative sum of every other element in the strided input array,
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusum/docs/repl.txt [hunk 2]
```diff
@@ -3,8 +3,8 @@
     Computes the cumulative sum of double-precision floating-point strided array
     elements.
 
-    The `N` and `stride` parameters determine which elements in `x` and `y` are
-    accessed at runtime.
+    The `N` and `stride` parameters determine which elements in the strided
+    input and output array are accessed at runtime.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusum/docs/repl.txt]:
```suggestion
    arrays are accessed at runtime.
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusum/examples/index.js [hunk 3]
```diff
@@ -18,20 +18,14 @@
...
-var round = require( '@stdlib/math/base/special/round' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var filledarrayBy = require( '@stdlib/array/filled-by' );
 var Float64Array = require( '@stdlib/array/float64' );
 var dcusum = require( './../lib' );
 
-var y;
-var x;
-var i;
+var x = filledarrayBy( 10, 'float64', discreteUniform( 0, 100 ));
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusum/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( 0, 100 ) );
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks @AuenKr for working on this, and thank you for being diligent and paying attention to detail on this. Really makes things easy on reviewers.

I think the only thing which needs to be updated is the `manifest.json` file. For an example, see https://github.com/stdlib-js/stdlib/blob/dc84fce54ee45d590860af4eb36845981342671f/lib/node_modules/%40stdlib/blas/ext/base/dfill/manifest.json. In short, we need to add specific build configurations for the addon vs benchmarks vs examples.

**Comment by AuenKr** [comment]
@kgryte, applied changes in `manifest.json`

File: lib/node_modules/@stdlib/blas/ext/base/dcusum/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, N, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideY, argv, 4 );
+
+	napi_value v;
+	status = napi_create_double( env, stdlib_strided_dcusum( N, sum, (double *)X, strideX, (double *)Y, strideY ), &v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusum/src/addon.c]:
```suggestion
	status = napi_create_double( env, stdlib_strided_dcusum( N, sum, X, strideX, Y, strideY ), &v );
```

Same comment as #1864 regarding casting.

File: lib/node_modules/@stdlib/blas/ext/base/dcusum/manifest.json [hunk 5]
```diff
@@ -1,42 +1,76 @@
...
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": [
-				"@stdlib/blas/ext/base/dcusumkbn"
-			]
-		}
-	]
+  "options": {},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusum/manifest.json]:
```suggestion
  "options": {
    "task": "build"
  },
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @AuenKr.

**Pull request merged**